Add MagicHome integration

ZNEGGE-SDK opened the pull request:
## Breaking Change:
I integrated my company's products into the home assistant. The API controls and retrieves the status of the device through remote HTTP, detailed functions and passes the test. If you have any questions, please feel free to ask.
![image](https://user-images.githubusercontent.com/17330822/68352548-baff0a80-0141-11ea-9d4e-16d9661d0933.png)
![image](https://user-images.githubusercontent.com/17330822/68352589-dbc76000-0141-11ea-8ad0-0a543546cd4e.png)
![image](https://user-images.githubusercontent.com/17330822/68352609-f1d52080-0141-11ea-9b33-ca226f6ed48b.png)
![image](https://user-images.githubusercontent.com/17330822/68352637-0d402b80-0142-11ea-9668-a137d3ee338c.png)

<!-- What is breaking and why we have to break it. Remove this section only if it was NOT a breaking change. -->

## Description:


**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#11110

## Example entry for `configuration.yaml` (if applicable):
```yaml
magichome:
    username: you account
    password: you password

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [x] Untested files have been added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Tests have been added to verify that the new code works.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

ZNEGGE-SDK committed:
Add MagicHome components file

homeassistant commented:
Hi @ZNEGGE-SDK, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#28581).

Once you do that we will be able to review and accept this pull request.

Thanks!

ZNEGGE-SDK committed:
update requirements_all.txt add magichome requirements

LIULi-VVET committed:
Fix cl check error

homeassistant commented:
Hi @LIULi-VVET, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#28581).

Once you do that we will be able to review and accept this pull request.

Thanks!

LIULi-VVET committed:
fix cl check error

homeassistant commented:
Hi @LIULi-VVET, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#28581).

Once you do that we will be able to review and accept this pull request.

Thanks!

LIULi-VVET committed:
Upgrade version and format

homeassistant commented:
Hi @LIULi-VVET, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#28581).

Once you do that we will be able to review and accept this pull request.

Thanks!

LIULi-VVET committed:
update magichome requirements_all

pvizeli commented on homeassistant/components/magichome/manifest.json:
```diff
@@ -0,0 +1,10 @@
+    "magichome==0.1.41"
+  ],
+  "dependencies": [],
+  "codeowners": []
```
Please add your github user there

ZNEGGE-SDK replied:
OK！

ZNEGGE-SDK committed:
Update manifest.json

LIULi-VVET committed:
Update github user

LIULi-VVET committed:
fix json file

LIULi-VVET committed:
Update CODEOWNERS

ZNEGGE-SDK committed:
Merge branch 'dev' into dev

LIULi-VVET committed:
update manifest.json

LIULi-VVET committed:
fix request Max retries bug

ZNEGGE-SDK committed:
Update manifest.json

LIULi-VVET committed:
update manifest.json and requirements_all.txt

LIULi-VVET committed:
Merge branch 'dev' of https://github.com/ZNEGGE-SDK/home-assistant into dev

# Conflicts:
#	homeassistant/components/magichome/manifest.json

fabaff commented on homeassistant/components/magichome/__init__.py:
```diff
@@ -0,0 +1,160 @@
+
+def setup(hass, config):
+    """Set up MagicHome Component."""
+    from magichome import MagicHomeApi
```
Import should be placed on the top of the file.

LIULi-VVET replied:
OK！

fabaff commented on homeassistant/components/magichome/__init__.py:
```diff
@@ -0,0 +1,160 @@
+from homeassistant.helpers import discovery
+from homeassistant.helpers.dispatcher import dispatcher_send, async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
```
Imports should be grouped.

fabaff commented on homeassistant/components/magichome/__init__.py:
```diff
@@ -0,0 +1,160 @@
+    "switch": "switch",
+    "socket": "socket",
+    "controller": "controller",
+}
```
This looks like a 1-to-1 mapping.

LIULi-VVET replied:
Yes, does this need to be modified?

springstan replied:
Yes this needs to be modified since 1-to-1 mapping is redundant.

springstan replied:
@LIULi-VVET this still needs to be changed

springstan replied:
I would recommend to use a list instead of a dictionary and remove line 70.

fabaff commented on homeassistant/components/magichome/__init__.py:
```diff
@@ -0,0 +1,160 @@
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(CONF_COMPANY, default="ZG001"): cv.string,
+                vol.Optional(CONF_PLATFORM, default="ZG001"): cv.string,
```
The documentation doesn't mention other platforms. If there are other platforms then there should be a list of supported manufactures.

LIULi-VVET replied:
We haven't negotiated with each other's platform yet. Actually, there are other platforms.
I'm writing here just to prepare, and maybe update the documentation by then

fabaff commented on homeassistant/components/magichome/__init__.py:
```diff
@@ -0,0 +1,160 @@
+    platform = config[DOMAIN][CONF_PLATFORM]
+
+    hass.data[DATA_MAGICHOME] = magichome
+    magichome.init(username, password, company, platform)
```
What happens if the credentials are wrong?

LIULi-VVET replied:
I have updated to deal with this issue in my package  . make an error log

LIULi-VVET committed:
update __init__  and update requirements version

LIULi-VVET committed:
fix wrong-import-order

LIULi-VVET committed:
fix wrong-import-order

LIULi-VVET committed:
fix wrong-import-order

LIULi-VVET committed:
fix too many blank lines

LIULi-VVET committed:
fix wrong-import-order

LIULi-VVET committed:
fix wrong-import-order

springstan commented:
@ZNEGGE-SDK tests are needed for all new integrations, therefore please add them :)

LIULi-VVET commented:
@springstan I'm so sorry, I can test locally, but the automatic detection fails after submitting

springstan commented:
@LIULi-VVET I am sorry, what do you mean by '_automatic detection fails_'?

ZNEGGE-SDK commented:
@springstan 
Like the one below
codecov / patch — 0% of diff hit (target 94.43%)
It display a red cross

LIULi-VVET commented:
@springstan
Like the one below
codecov / patch — 0% of diff hit (target 94.43%)
It display a red cross

ZNEGGE-SDK commented:
> @ZNEGGE-SDK tests are needed for all new integrations, therefore please add them :)

I am sorry . What do you mean by "add them"?

springstan commented:
@ZNEGGE-SDK like it is mentioned in the [devlopment checklist](https://developers.home-assistant.io/docs/en/creating_component_code_review.html#7-tests):

> 7. Tests
Strongly consider adding tests for your component to minimize future regressions.

ZNEGGE-SDK commented:
> @ZNEGGE-SDK like it is mentioned in the [devlopment checklist](https://developers.home-assistant.io/docs/en/creating_component_code_review.html#7-tests):
> 
> > 1. Tests
> >    Strongly consider adding tests for your component to minimize future regressions.

OK!

LIULi-VVET committed:
Add tests components

LIULi-VVET committed:
updata homeassistant/components/magichome/__init__.py

LIULi-VVET committed:
update test_init.py

springstan commented:
@LIULi-VVET thanks for adding these tests! It is really appreciated 👍
Please take a look at the errors on [Azure Devops](https://dev.azure.com/home-assistant/Home%20Assistant/_build/results?buildId=19576&view=results) :)

LIULi-VVET commented:
> @LIULi-VVET thanks for adding these tests! It is really appreciated 👍
> Please take a look at the errors on [Azure Devops](https://dev.azure.com/home-assistant/Home%20Assistant/_build/results?buildId=19576&view=results) :)

Thank you! I know something went wrong and I'm working on it

LIULi-VVET committed:
fix flake8 error

LIULi-VVET committed:
update exec black test_init.py

springstan commented:
Just a quick tip,  you need to run `python3 -m script.gen_requirements_all`, see [Azure DevOps](https://dev.azure.com/home-assistant/Home%20Assistant/_build/results?buildId=19581&view=logs&j=f36f3843-8e6d-51cb-0e3a-71707ab37859&t=e02bf7dd-8cde-5cf1-60a6-6b38a38544a6&l=31)

LIULi-VVET commented:
> Just a quick tip, you need to run `python3 -m script.gen_requirements_all`, see

Thank you !  I try it , but I don't know why it says 'Errors while importing: The manifest for integration magichome is invalid.' 
I still look for the problem, if you know how to solve it, can you give me a hint?

LIULi-VVET commented:
> Just a quick tip, you need to run `python3 -m script.gen_requirements_all`, see [Azure DevOps](https://dev.azure.com/home-assistant/Home%20Assistant/_build/results?buildId=19581&view=logs&j=f36f3843-8e6d-51cb-0e3a-71707ab37859&t=e02bf7dd-8cde-5cf1-60a6-6b38a38544a6&l=31)

Last I execute `script/lint`  Got almost the same error "Unable to import 'magichome' (import-error)"

LIULi-VVET committed:
upload level up

springstan commented on tests/components/magichome/test_init.py:
```diff
@@ -0,0 +1,563 @@
+import pytest
+
+from homeassistant import core
+from homeassistant.component import magichome as MagicHomeApi
```
Just a small typo: `components` instead of `component` 😉

LIULi-VVET replied:
OK！ Thank

LIULi-VVET committed:
fix import error

LIULi-VVET committed:
update test_init.py

LIULi-VVET committed:
update test_init.py

LIULi-VVET committed:
update test_init.py

LIULi-VVET committed:
update test_init.py

LIULi-VVET committed:
update test_init.py

LIULi-VVET committed:
update test_init.py

LIULi-VVET committed:
update test_init.py

LIULi-VVET committed:
update `black test_init.py`

springstan commented:
@LIULi-VVET a small hint: you always need to format your code with black; by running `black --fast homeassistant` inside of your dev environment.

ZNEGGE-SDK commented:
> @LIULi-VVET a small hint: you always need to format your code with black; by running `black --fast homeassistant` inside of your dev environment.

 OK! Thank you!

LIULi-VVET committed:
fix test_init.py

LIULi-VVET committed:
fix test_init.py

LIULi-VVET commented:
> @LIULi-VVET a small hint: you always need to format your code with black; by running `black --fast homeassistant` inside of your dev environment.

I'm so sorry. For some reason I edited in Notepad, resulting in many submissions.

LIULi-VVET commented:
@springstan I'm so sorry. For some reason I edited in Notepad, resulting in many submissions.

LIULi-VVET commented:
@springstan oh! I reinstalled the development environment when before modifying

LIULi-VVET committed:
fix test_init.py

LIULi-VVET committed:
update test_init.py

LIULi-VVET committed:
update test_init.py

LIULi-VVET commented:
@springstan  I submitted the test file and see if there are any problems

springstan commented:
@LIULi-VVET the tests you added look good so far. Codecoverage is still failing since the other platforms are not covered yet, e.g. light, scene, socket and so on.

LIULi-VVET commented:
@springstan OK ! Thank you !

LIULi-VVET committed:
update 0.1.46

LIULi-VVET committed:
update

LIULi-VVET committed:
update

LIULi-VVET committed:
update

LIULi-VVET committed:
update

LIULi-VVET committed:
update

LIULi-VVET committed:
upload 0.1.47

LIULi-VVET committed:
update

LIULi-VVET committed:
update

LIULi-VVET commented:
OK！Thank you very much！






------------------&nbsp;Original&nbsp;&nbsp;------------------
From:&nbsp;"springstan"<notifications@github.com&gt;;
Send Time:&nbsp;2019/12/24/ AM 3:31
To:&nbsp;"home-assistant/home-assistant"<home-assistant@noreply.github.com&gt;;
CC:&nbsp;"SmartAteam-Verter"<974870235@qq.com&gt;;"Mention"<mention@noreply.github.com&gt;;
Subject:&nbsp;Re: [home-assistant/home-assistant] Add MagicHome components file (#28581)




@springstan commented on this pull request.
 
 
In homeassistant/components/magichome/__init__.py:
 &gt; +DOMAIN = "magichome" +DATA_MAGICHOME = "data_magichome" + +SIGNAL_DELETE_ENTITY = "magichome_delete" +SIGNAL_UPDATE_ENTITY = "magichome_update" + +SERVICE_FORCE_UPDATE = "force_update" +SERVICE_PULL_DEVICES = "pull_devices" + +MAGICHOME_TYPE_TO_HA = { +    "light": "light", +    "scene": "scene", +    "switch": "switch", +    "socket": "socket", +    "controller": "controller", +}  
Yes this needs to be modified since 1-to-1 mapping is redundant.
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

LIULi-VVET commented:
@springstan Hi ! I submitted the test file and see if there are any problems

springstan commented on homeassistant/components/magichome/manifest.json:
```diff
@@ -0,0 +1,10 @@
+{
+  "domain": "magichome",
+  "name": "MagicHome",
+  "documentation": "https://www.home-assistant.io/components/magichome",
```
Replace `components` with `integrations`

LIULi-VVET committed:
Update _init_.py

Reduction of device type

LIULi-VVET committed:
Update _init_.py

LIULi-VVET committed:
Update manifest.json

LIULi-VVET committed:
update __init__.py

LIULi-VVET commented:
@frenck 
Is this the document?(https://github.com/home-assistant/home-assistant.io/pull/11110)

frenck commented:
Thanks, @ZNEGGE-SDK! 👍 
I've updated the OP to include that link (as per PR template).

LIULi-VVET commented:
@frenck Thank you very much !

balloob committed:
Bumped version to 0.106.0b0

frenck committed:
Merge branch 'dev' into rc

dmulcahey committed:
Bump ZHA quirks and add skip configuration support (#31982)

* add skip configuration
* Bump quirks
* add skip configuration to FakeDevice

balloob committed:
Only check frontend for safe mode if frontend wanted to be loa… (#31969)

* Only check frontend for safe mode if frontend wanted to be loaded

* Update test

marthoc committed:
Fix bug in ecobee integration (#32008)

* Bump python-ecobee-api to 0.2.1

* Update log messages for clarity

* Update requirements_all

tsvi committed:
Don't return coroutine in DLNA/DMR service handler (#32011)

Fixes #32010

jjlawren committed:
Fix Plex sensor title handling (#31973)

* Fix Plex sensor title parsing

* Revert episode year for now

MatthewFlamm committed:
Add missing name to logging in DataUpdateCoordinator (#32023)

balloob committed:
Fix hue test

frenck committed:
Remove deprecated Hue options (fixes CI) (#32027)

cgtobi committed:
Bump pyatmo to 3.2.4 (#32018)

* Bump pyatmo to 3.2.3

* Bump pyatmo version to 3.2.4

* Update requirements_test_all.txt

balloob committed:
Fix recursion bug (#32009)

* Fix recursion bug

* Remove shield

frenck committed:
Add minimal version contrain to urllib3 (#32031)

balloob committed:
Updated frontend to 20200220.0 (#32033)

balloob committed:
Bumped version to 0.106.0b1

balloob committed:
Updated frontend to 20200220.1 (#32046)

bdraco committed:
Fix i/o in august camera async image update (#32044)

* Fix i/o in august camera image update

* Address review comments

bieniu committed:
Bump brother to 0.1.6 (#32054)

arsaboo committed:
Add additional logging to rest sensor (#32068)

jjlawren committed:
Add ConfigEntryNotReady exception to Plex (#32071)

ziv1234 committed:
Enhance Dynalite Integration after review (#31760)

* fixes per Martin Hjelmare

* pylint fix

* final fixes per request

* fixed unit tests for new config flow

* Added unit-tests to increase coverage. at 97% now

* Added unit tests for 100% coverage of component

* removed configured_host function and updated config_flow unit tests

* added a pylint directive since it tells me by mistake DOMAIN is not used

* fixed path (removed __init__)

* Update homeassistant/components/dynalite/light.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/dynalite/light.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* fixed the test as we moved from schedule_update_... to async_schedule

* Update homeassistant/components/dynalite/bridge.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* removed context from config_flow
changed test_init to use the core methods

* moved test_light to also use the core interfaces

* moved to config_entries.async_unload

* additional fixes for the tests

* pylint fix and removed unnecessary code

* Update tests/components/dynalite/test_light.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/dynalite/test_light.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/dynalite/test_light.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/dynalite/test_light.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/dynalite/test_light.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/dynalite/test_light.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/dynalite/test_light.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/dynalite/test_light.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/dynalite/test_light.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* added break in loop

* removed last mock_coro reference
pylint fix

* added coverage for try_connect

* added check for a successful connection before bridge.async_setup succeeds
also added a "nowait" config option (default False) that avoids this check

* changed log level

* fixed accidental chmod I did

* fixed accidental change

* not storing config in bridge

* not patching asyncio

* moved CONFIG_SCHEMA into component

* moved all logs to start capitalized (and revised some of them)

* moved test_config_flow to not patch the DynaliteBridge

* also took DynaliteBridge patching out of test_init

* removed NO_WAIT

* fixes to SCHEMA

* changed _ in multi-word CONF
moved imports to component const.py

* removed tries

* removed redundant tests

* fixed some small change i broke in the library. only version update

* fixed rewuirements

* Update tests/components/dynalite/test_config_flow.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/dynalite/test_light.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update tests/components/dynalite/test_config_flow.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* removed HIDDEN_ENTITY
removed hass in test fixture

* black fixes

* removed final piece of hidden_entity from light
fix in the library
updated config flow so if the entry is already set but with a different config, calls async_update_entry

* removed DATA_CONFIGS - no longer necessary

* pylint fixes

* added coverage

* use abort in config_flow

* test update

* removed logs

* test that update actually updates the entry

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

balloob committed:
Remove YAML config from Ring integration (#32039)

balloob committed:
Bumped version to 0.106.0b2

CoMPaTech committed:
Fix Plugwise climate issues for new firmware #32080 (#32109)

* Fix Plugwise climate issue firmare 3.1.11 (#32080)

* Submit checklist actions

kit-klein committed:
0.106 Beta - provide correctly formatted placeholders (#32119)

balloob committed:
Properly define depenency for Scrape integration on Rest integ… (#32136)

raman325 committed:
Fix vizio bug to use 'get' to get volume_step since it is optional (#32151)

* use get to get volume_step since it is optional

* always set volume_step to default in options and data if its not included

raman325 committed:
Improve Vizio fix to avoid KeyError (#32163)

* dont set default volume_step on import check

* ensure config entry data['volume_step'] is set

* consolidate entry update during entry setup

balloob committed:
Bumped version to 0.106.0b3

balloob committed:
Bumped version to 0.106.0b4

kit-klein committed:
Dedup and clarify imported konnected config flows (#32138)

* dedup config flows

* use default (imported) options until user goes thru options flow

* address pr feedback

* correct key used to distinguish pro model

michaelarnauts committed:
Bump python-tado to 0.3.0 (#32186)

pinkywafer committed:
Bump socialbladeclient to 0.5 (#32191)

balloob committed:
Fix Arlo doing I/O in event loop (#32190)

bramkragten committed:
Updated frontend to 20200220.3 (#32189)

balloob committed:
Update translations

balloob committed:
Bumped version to 0.106.0b5

bramkragten committed:
Updated frontend to 20200220.4 (#32205)

bachya committed:
Fix error where SimpliSafe websocket would disconnect and not reconnect (#32199)

* Fix error where SimpliSafe websocket would disconnect and not reconnect

* Await

frenck committed:
Bumped version to 0.106.0

frenck committed:
Merge pull request #32211 from home-assistant/rc

0.106.0

emontnemery committed:
Add missing translations for light actions (#32216)

dupondje committed:
Fix DSMR 5 (#32233)

DSMR 5 was broken because some wrong if.
if dsmr_version in ("5B"):
-> this checks dsmr_version against 5 and B. Not if its 5B.

bachya committed:
Bump simplisafe-python to 9.0.0 (#32215)

bachya committed:
Bump simplisafe-python to 9.0.2 (#32273)

Kane610 committed:
deCONZ - Race condition on slower systems (#32274)

When battery sensors gets created before other platforms loading deconz sensors gets created first the other platform would not create entities related to those battery sensors

lociii committed:
Mark clients away if they have never been seen. (#32222)

balloob committed:
Catch more Hue errors (#32275)

balloob committed:
Bumped version to 0.106.1

balloob committed:
Fix dsmr test

dgomes committed:
Bump pyipma dependency (fixes bug in 0.106) (#32286)

* Bump version

* Bump PyIPMA version

balloob committed:
Merge pull request #32282 from home-assistant/rc

0.106.1

bramkragten committed:
Updated frontend to 20200220.5 (#32312)

Kane610 committed:
UniFi - Temporary workaround to get device tracker to mark cli… (#32321)

balloob committed:
revent saving/deleting Lovelace config in safe mode (#32319)

balloob committed:
Bumped version to 0.106.2

bdraco committed:
Ensure rest sensors are marked unavailable when http requests… (#32309)

balloob committed:
Merge pull request #32327 from home-assistant/rc

0.106.2

elmurato committed:
Fixed TypeError with old server versions (#32329)

mezz64 committed:
Bump pyeight to 0.1.4 (#32363)

balloob committed:
Add coronavirus integration (#32413)

* Add coronavirus integration

* Update homeassistant/components/coronavirus/manifest.json

Co-Authored-By: Franck Nijhof <git@frenck.dev>

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

balloob committed:
Bumped version to 0.106.3

balloob committed:
Merge pull request #32414 from home-assistant/rc

0.106.3

bramkragten committed:
Update azure-pipelines-release.yml for Azure Pipelines

balloob committed:
Coronavirus updates (#32417)

* Sort countries alphabetically

* Update sensor name

* Add migration to stable unique IDs

* Update sensor.py

mezz64 committed:
Catch Eight Sleep API errors, don't round None type (#32410)

* Catch API errors, don't round None type

* Specify error type

balloob committed:
Bumped version to 0.106.4

balloob committed:
Add unique ID to coronavirus (#32423)

balloob committed:
Merge pull request #32424 from home-assistant/rc

0.106.4

bdraco committed:
Properly define dependency for pvoutput integration on rest in… (#32435)

balloob committed:
Filter out duplicate logbook states (#32427)

balloob committed:
Remove hassfest blacklisted rest (#32441)

* Remove blacklisted deps from hassfest deps

* Whitelist all internal integrations

Kane610 committed:
UniFi - Fix websocket bug (#32449)

axilleas committed:
Fix pushover's ATTR_RETRY env variable typo (#32440)

alandtse committed:
Fix too many device tracker updates in log for Tesla (#32426)

* Fix Tesla too many device tracker updates in log

* Empty commit to re-trigger build

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

balloob committed:
Bumped version to 0.106.5

balloob committed:
Merge pull request #32457 from home-assistant/rc

0.106.5

LIULi-VVET committed:
Merge remote-tracking branch 'upstream/master' into dev
